lib/manager/components/transform/CustomCSVForm.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import React from 'react'
+import { Button } from 'react-bootstrap'
+
+import { getComponentMessages } from '../../../common/util/config'
+
+type Props = {
+  csvData: ?string,
+  inputIsSame: boolean,
+  onChangeCsvData: (SyntheticInputEvent<HTMLInputElement>) => void,
+  onSaveCsvData: () => void,
+  type: string
+}
+const CustomCSVForm = (props: Props) => {
+  const { csvData, inputIsSame, onChangeCsvData, onSaveCsvData, type } = props
+
+  const numLines = !csvData ? 0 : csvData.split(/\r*\n/).length
+  const messages = getComponentMessages('CustomCSVForm')
+
+  let header
+  switch (type) {
+    case 'ReplaceFileFromString':
+      header = messages('addCsvData')
+      break
+    case 'AddCustomFile':
+      header = messages('addCustomCsvData')
+      break
+    case 'PreserveCustomFields':
+      header = messages('addCsvWithCustomFields')
+      break
```

Comment:
Since this form is implemented in these 3 different parent components can't we just pass in the string as a prop?

Reply:
You're right that was silly, adjusted that in https://github.com/ibi-group/datatools-ui/pull/950/commits/2909ffe33ac841874be7670c43c6d44991d41254